### PR TITLE
Feature/mPMT LED generator

### PR DESCRIPTION
**Pull request by kmtsui**
New primary generator for mPMT LED
- Dynamically load PMT positions after detector construction, then calculate source position and direction based on mPMT id and LED id
- Allow short change in source direction for uncertainty estimation
- Source position and direction are saved as vertex and beam direction (first particle track direction) in output

Plus a small bug fix of saving multiple vertices information in radon generator

### CODE REVIEW

File: WCSim.mac
Comment (tdealtry) [inline WCSim.mac:358]: Naive question here. I thought the LED mPMT proposal was to replace 5 (of the 19) 3" PMTs each with 8 LEDs = 40 LEDs in total. Why the number 12 here?

File: src/WCSimPrimaryGeneratorAction.cc
Comment (tdealtry) [inline src/WCSimPrimaryGeneratorAction.cc]: I think this can be deleted? e.g. `zpos` is already defined above

File: src/WCSimPrimaryGeneratorAction.cc
Comment (tdealtry) [inline src/WCSimPrimaryGeneratorAction.cc]: Can you do this kind of stuff only for the first event?

File: src/WCSimPrimaryGeneratorAction.cc
Comment (tdealtry) [inline src/WCSimPrimaryGeneratorAction.cc]: Think you're doing this via mac file now, so please remove

**Review by tdealtry — CHANGES_REQUESTED** [review]
Thanks @kmtsui Looks good. Few minor things, while we wait on me to fix the CI

**Comment by tdealtry** [comment]
Turns out the CI is actually working 🥳 So just those couple of minor comments to address before merging

File: WCSim.mac
Comment (kmtsui) [reply WCSim.mac:358]: This is another set of calibration LEDs which will be placed between the small PMTs

Comment (kmtsui) [reply WCSim.mac:358]: The LEDs are the green dots in the figure.
image

File: src/WCSimPrimaryGeneratorAction.cc
Comment (kmtsui) [reply src/WCSimPrimaryGeneratorAction.cc]: Done

File: src/WCSimPrimaryGeneratorAction.cc
Comment (kmtsui) [reply src/WCSimPrimaryGeneratorAction.cc]: Done

File: WCSim.mac
Comment (tdealtry) [reply WCSim.mac:358]: Ahh I see, thanks. The difference between "mPMT LED" and "LED-mPMT". Not confusing at all... 
They're the names we use in the collaboration, so we have to use them. But what we can do is be explicit to avoid confusion. Something like
```
# Or inject photon from mPMT LED
# (Note that these are LEDs positioned in the standard mPMT modules.
#  They are not LEDs in the LED-mPMT modules)
```

**Comment by tdealtry** [comment]
Thanks @kmtsui I've just set the CI off again, so happy for you to merge if it comes up green, and when you resolve the last comment. Thanks

**Comment by tdealtry** [comment]
Ahh I always forget this. Rerunning a CI job uses the old, rather than new, CI workflow. You'll have to push the fix to the comment to set it running.

File: WCSim.mac
Comment (kmtsui) [reply WCSim.mac:358]: done

**Review by tdealtry — APPROVED** [review]

**Comment by tdealtry** [comment]
Thanks @kmtsui

**Pull request merged**
